src/main/java/com/dataguise/saas/JWTLoginFilter.java

```diff
@@ -0,0 +1,51 @@
+package com.dataguise.saas;
+
+import com.dataguise.saas.dto.LoginDTO;
+import com.dataguise.saas.service.JWTService;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Collections;
+
+public class JWTLoginFilter extends AbstractAuthenticationProcessingFilter {
+
+    public JWTLoginFilter(String url, AuthenticationManager authManager) {
+        super(new AntPathRequestMatcher(url));
+        setAuthenticationManager(authManager);
+    }
+
+    @Override
+    public Authentication attemptAuthentication(
+            HttpServletRequest req, HttpServletResponse res)
+            throws AuthenticationException, IOException, ServletException {
+        LoginDTO creds = new ObjectMapper()
+                .readValue(req.getInputStream(), LoginDTO.class);
+        return getAuthenticationManager().authenticate(
+                new UsernamePasswordAuthenticationToken(
+                        creds.getUsername(),
+                        creds.getPassword(),
+                        Collections.emptyList()
+                )
+        );
+    }
+
+    @Override
+    protected void successfulAuthentication(
+            HttpServletRequest req,
+            HttpServletResponse res, FilterChain chain,
```

Comment:
**Reporter**: Checkstyle    
    **Rule**: com.puppycrawl.tools.checkstyle.checks.FinalParametersCheck
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/JWTLoginFilter.java L45
    
    Parameter chain should be final.
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a1914175603>*